### PR TITLE
Updated to use DBConnection pool

### PR DESCRIPTION
**Pull request by Darchie4**
Updated to use DBConnection pool in all cases.
Refactored Tokenizer

### CODE REVIEW

File: src/main/java/dk/sdu/se_f22/searchmodule/infrastructure/DelimiterSettings.java [hunk 1]
```diff
@@ -28,7 +28,8 @@ private void updateDelimitersToDatabaseState() throws SQLException {
     }
 
     private ResultSet getAllDelimitersFromDatabase() throws SQLException {
-        PreparedStatement stmt = DBConnection.getConnection().prepareStatement("SELECT * FROM searchtokendelimiters");
+        Connection dbConnection = DBConnection.getPooledConnection();
```
Comment (mathiasgredal) [inline src/main/java/dk/sdu/se_f22/searchmodule/infrastructure/DelimiterSettings.java:31]: Badness 10000

File: src/main/java/dk/sdu/se_f22/searchmodule/infrastructure/DelimiterSettings.java [hunk 2]
```diff
@@ -72,14 +73,17 @@ private boolean executeStatementIfDelimIsInDelimiters(String delim, PreparedStat
...
                 return true;
             }
         }
+        stmt.close();
         return false;
     }
 
     private PreparedStatement prepareDeleteStatement(String sql, String delim) throws SQLException {
-        PreparedStatement stmt = DBConnection.getConnection().prepareStatement(sql);
+        Connection dbConnection = DBConnection.getPooledConnection();
```
Comment (mathiasgredal) [inline src/main/java/dk/sdu/se_f22/searchmodule/infrastructure/DelimiterSettings.java:85]: Badness 10000

File: src/test/java/dk/sdu/se_f22/searchmodule/infrastructure/TokenizationTest.java [hunk 3]
```diff
@@ -41,8 +42,8 @@ void tokenizeWithSettings() {
 
     @Test
     void tokenizeWithNoSettings(){
-        try {
-            PreparedStatement stmt = DBConnection.getConnection().prepareStatement("DELETE FROM searchtokendelimiters");
+        try (Connection connection = DBConnection.getPooledConnection()){
```
Comment (mathiasgredal) [inline src/test/java/dk/sdu/se_f22/searchmodule/infrastructure/TokenizationTest.java:45]: Yay

**Review by mathiasgredal — CHANGES_REQUESTED** [review]

File: src/main/java/dk/sdu/se_f22/searchmodule/infrastructure/DelimiterSettings.java (hunk 2 above)
Comment (Darchie4) [reply src/main/java/dk/sdu/se_f22/searchmodule/infrastructure/DelimiterSettings.java]: Fixed

File: src/test/java/dk/sdu/se_f22/searchmodule/infrastructure/TokenizationTest.java (hunk 3 above)
Comment (Darchie4) [reply src/test/java/dk/sdu/se_f22/searchmodule/infrastructure/TokenizationTest.java:45]: Not fixed

**Pull request merged**